src/tests/Common/XUnitLogChecker/XUnitLogChecker.csproj

```diff
@@ -7,7 +7,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <PublishAot Condition="'$(NativeAotSupported)' == 'true'">true</PublishAot>
     <SuppressTrimAnalysisWarnings>false</SuppressTrimAnalysisWarnings>
-    <LinkerFlavor Condition="'$(CrossBuild)' == 'true' and '$(TargetsLinux)' == 'true'">lld</LinkerFlavor>
+    <LinkerFlavor Condition="('$(__TestBuildMode)' == 'nativeaot' or '$(CrossBuild)' == 'true') and '$(TargetsLinux)' == 'true'">lld</LinkerFlavor>
```

Comment:
This indicates our test build isn't passing `CrossBuild` down correctly. I think we are missing `$(crossArg)` in the `./src/tests/build.sh` invocations in the templates. I think this explains the `-lz` mystery issue as well.

Reply:
Done.